FireboltNETSDK/FireboltClient2.cs

```diff
@@ -105,32 +105,59 @@ public override async Task<ConnectionResponse> ConnectAsync(string? engineName,
         await EstablishConnection();
         // Connecting to system engine by default
         var result = await GetSystemEngineUrl(_account);
-        _connection.EngineUrl = result.engineUrl;
-        if (engineName == null)
+        if (result.engineUrl != null)
         {
-            return new ConnectionResponse(result.engineUrl!, database, true);
+            string[] urlParts = result.engineUrl.Split('?');
+            _connection.EngineUrl = urlParts[0];
+            string? accountId = _connection.AccountId; // initializes InfraVersion and connection.accountId
+            if (urlParts.Length > 1)
+            {
+                ProcessParameters(new FireboltConnectionStringBuilder(_connection.ConnectionString), ExtractParameters(urlParts[1]), false);
+            }
+        }
+        if (engineName == null || FireboltConnection.SYSTEM_ENGINE.Equals(engineName))
+        {
+            return ConnectToSystemEngine(_connection.InfraVersion, database);
         }
         // Engine is specified
-        if (database == string.Empty)
+        return ConnectToCustomEngine(_connection.InfraVersion, engineName, database);
+    }
+
+    private ConnectionResponse ConnectToSystemEngine(int infraVersion, string database)
+    {
+        Execute("select 1"); // needed to get the InfraVersion back
+        if (_connection.InfraVersion == 2)
         {
-            // If no db provided - try to fetch it
-            database = GetEngineDatabase(engineName) ?? throw new FireboltException($"Engine {engineName} is attached to a database current user can not access");
+            _protocolVersion = "2.1";
```

Comment:
Let's make `2.1` a const somewhere at the top. Otherwise we'll almost certainly forget to update it in one of the cases below when the time comes to bump it to `2.2`.